app/src/swap/handle_sign_transaction.c

```diff
@@ -93,7 +93,7 @@ parser_error_t check_swap_conditions(const parser_context_t *ctx) {
                                                                 ctx->tx_obj->callIndex.moduleIdx,
                                                                 ctx->tx_obj->callIndex.idx));
 
-    if (strncmp(tmp_str, &valid_tx_method[0], strlen(valid_tx_method)) != 0) {
+    if (strlen(valid_tx_method) != strlen(tmp_str) || strncmp(tmp_str, &valid_tx_method[0], strlen(valid_tx_method)) != 0) {
```

Comment:
I know that tmp_str won't be bigger than 80 but it would be good to replace this strlen by strnlen

Reply:
Paranoia fixed 👍🏻 😂